lib/msf/core/modules/external/bridge.rb

```diff
@@ -72,6 +72,10 @@ def send(message)
         else
           raise "Error running module #{self.path}"
         end
+      rescue ::Exception => e
+        raise handle_exception(e)
+      else
+        e
       end
```

Comment:
In this scenario I don't believe we want to catch `::Exception` - https://thoughtbot.com/blog/rescue-standarderror-not-exception.

Looks like the feedback of providing a default implementation in the base class is required still; This will avoid the potential of a `NoMethodError` occurring when the overriding class doesn't provide a `handle_exception` implementation:

```suggestion
      rescue => e
        raise handle_exception(e)
      end
      
      def handle_exception(e)
        e
      end
```

Reply:
Ok, got that added in. That makes more sense now after reading that link 👍 , thanks!